src/server/game/Spells/Spell.cpp

```diff
@@ -2531,6 +2531,13 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
 
             caster->DealSpellDamage(&damageInfo, true);
 
+            // Sparring Checks
```

Comment:
DealSpellDamage calls DealDamage which you already modified with this same check, double checking here is unnecessary

Reply:
not exactly. I am modifiying the damageInfo damage here to make the spell damage log packet also send zero damage because the spell log damage packet is responsible to handle the clientside health damage.
So if I don't do this here, the core will indeed have the correct health, while the spell damage will still deal damage clientside.